### DISCUSSION TITLE
Can I use the "lamellar" and "diameq" functions at the same time?

### DISCUSSION THREAD

**Discussion by aksiteray** [neperfepx/neper#265] [Q&A] [answered]
I am hoping to create sub-grain with "diameq" function and would like to apply lamellar specifically to each sub-grain, not the entire grain with same lamellar. Is this possible? If so, how it is used?

I know that each can be done individually as thought at "https://neper.info/generation.html#generalities": 
```
$ neper -T -n 200::from_morpho -morpho "gg::lamellar(w=0.02)"
$ neper -T -n 200::1000*size -morpho 'gg::diameq:dirac(1),sphericity:lognormal(0.145,0.03,1-x)'
```

**Comment by rquey** [comment, marked as answer]
Yes, you can do 3-scale:

```
$ neper -T -n 10::100*size::from_morpho -morpho 'gg::diameq:dirac(1),1-sphericity:lognormal(0.145,0.03)::lamellar(w=0.02)'
```
![image](https://user-images.githubusercontent.com/2659376/125762404-8fb4927e-7f4f-408a-9b50-694de6832e75.png)

**Reply by zszAachen** [reply]
> Yes, you can do 3-scale:
> 
> ```
> $ neper -T -n 10::100*size::from_morpho -morpho 'gg::diameq:dirac(1),1-sphericity:lognormal(0.145,0.03)::lamellar(w=0.02)'
> ```
> 
> ![image](https://user-images.githubusercontent.com/2659376/125762404-8fb4927e-7f4f-408a-9b50-694de6832e75.png)

Hello Rquey,

Is it possible to use the "diameq" function in the third layer? According to your code, the width of lamella is a fixed value (w=0.02). 

Many thanks in advance,
zszAachen

**Reply by rquey** [reply]
I'm not sure what exactly you want to do, but yes, just put it there.

**Reply by zszAachen** [reply]
Hi Rquey,

Okay, in the third layer, I want absolute lamella width(w) to follow a lognormal distribution, rather than having the same lamellar width(w=0.02) throughout the grain. Is it possible? If you are still confused, please let me know.
By the way, please tell me where I can install Neper version 4.2.1-4.

Thanks in advance,
zszAachen

**Reply by aksiteray** [reply]
> Yes, you can do 3-scale:
> 
> ```
> $ neper -T -n 10::100*size::from_morpho -morpho 'gg::diameq:dirac(1),1-sphericity:lognormal(0.145,0.03)::lamellar(w=0.02)'
> ```
> 
> ![image](https://user-images.githubusercontent.com/2659376/125762404-8fb4927e-7f4f-408a-9b50-694de6832e75.png)

Much obliged! Thank you very much for the response.

**Reply by rquey** [reply]
> Okay, in the third layer, I want absolute lamella width(w) to follow a lognormal distribution, rather than having the same lamellar width(w=0.02) throughout the grain. Is it possible?

Statistical distributions are not supported here, but you could provide the successive values (obtained from the lognormal distribution) as follows (here for a gaussian distribution):
``` 
$ gsl-randist 1 100 gaussian 0.01 | awk '{print $1+0.03}' > w
$ more w | tr '\n' ':' > w2
$ neper -T -n from_morpho -morpho "lamellar(w=`more w2`)"

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.2.1-20-devel
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 8 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2021, and GNU GPL'd, by Romain Quey.
Info   : Loading initialization file `/home/rquey/.neperrc'...
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] -tesrformat ascii
Info   : [com line] -n from_morpho -morpho
         lamellar(w=0.033064:0.0257561:0.0279545:0.0257126:0.0430547:0.0364748:0.0419423:0.0125953:0.0326071:0.0273217:0.0268596:0.0303366:0.0361479:0.0180828:0.0100665:0.0360152:0.0363941:0.0239691:0.0318327:0.0335533:0.0319827:0.0479284:0.0290007:0.0159883:0.018912:0.0217268:0.0404064:0.0269296:0.0367817:0.0215683:0.0257543:0.0234963:0.0152244:0.0355876:0.023519:0.0307313:0.0357358:0.0209635:0.0565081:0.0309329:0.0491335:0.0146337:0.0198406:0.030796:0.0358137:0.0293067:0.0410241:0.0239537:0.0211872:0.0446122:0.0395841:0.0309211:0.028415:0.0252453:0.0386699:0.0345934:0.0279846:0.0388297:0.0371951:0.0237042:0.033508:0.0449763:0.0225829:0.0422549:0.0217213:0.0326907:0.0398942:0.0297553:0.0233244:0.0323905:0.00944:0.0364767:0.0126258:0.0376945:0.0318625:0.012624:0.0194299:0.0381189:0.0274221:0.0334383:0.0255257:0.0317828:0.0240271:0.0350023:0.0289161:0.0340957:0.0265845:0.0305286:0.0223481:0.0316438:0.0292969:0.029437:0.0301002:0.025946:0.0256999:0.0145109:0.0348887:0.0250827:0.026689:0.0252124:)
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Running tessellation...
Info   :   - Generating crystal orientations...
Info   : Writing results...
Info   :     [o] Writing file `nfrom_morpho-id1.tess'...
Info   :     [o] Wrote file `nfrom_morpho-id1.tess'.
Info   : Elapsed time: 0.009 secs.
========================================================================
```
